### PR TITLE
Implement capture animation + minor changes (see description)

### PR DESCRIPTION
**Pull request by presstabstart**
This branch successfully implements the capture animation. It's a prototype right now, and open to discussion.

Since I messed up my pull requests this one includes the previous 2, here are the descriptions on the changes:

Change the combat target selection menu for items to that of the the technique target selection menu:
-------------------------------------------------------------------------------
This allows us to select hostile monsters. Also adds a stud capture animation.

Added parameter dictionary to the .use methods for the technique/item objects and new catching algorithm:
-------------------------------------------------------------------------------
Added placeholder for capture animation, and implemented a capture algorithm where the tuxemon shakes the ball against a shake check.

The return value for items and techniques is now a DICTIONARY to allow for misc parameters. Note that I didn't include it in this commit but this solves the problem of not being able to determine whether or not to tackle, etc set out in the TODO of states/combat/combat.py, around line 550.

Limitations & Bugs
-------------------------------------------------------------------------------
- Capture animation is pretty ugly. This shouldn't be a problem, since it's just a placeholder.
- Since there aren't any target restrictions yet, the player can capture their own tuxemon, effectively duplicating their list of tuxemon.

### CODE REVIEW

**Comment by presstabstart** [comment]
I've now removed the other PRs and did some clean ups. I'll get to changing the return value for .use() to just a simple dictionary now and adding to en_US.json

File: tuxemon/core/states/combat/combat.py [hunk 1]
```diff
@@ -558,20 +560,31 @@ def perform_action(self, user, technique, target=None):
...
+                    action_time = result["num_shakes"] + 1.8
+                    if result["success"]: # end combat right here
+                        self.task(self.end_combat, action_time + 0.5) # Display 'Gotcha!' first.
+                        self.task(partial(self.alert, trans('gotcha')), action_time)
+                        self.alert(message)
+                        self._animation_in_progress = True
+                        return
+
+                if result["success"]:
+                	message += "\n" + trans('item_success')
```
Comment (bitcraft) [inline tuxemon/core/states/combat/combat.py]: mix of tabs/space

File: tuxemon/core/states/combat/combat.py [hunk 2]
```diff
@@ -558,20 +560,31 @@ def perform_action(self, user, technique, target=None):
                 self._damage_map[target].add(user)
 
             else:  # assume this was an item used
-                if result:
-                    message += "\n" + trans('item_success')
+                if result["name"] == "capture":
```
Comment (bitcraft) [inline tuxemon/core/states/combat/combat.py]: Traceback (most recent call last):
  File "C:/Users/Leif/PyCharmProjects/Tuxemon/tuxemon/main.py", line 38, in <module>
    main()
  File "C:\Users\Leif\PyCharmProjects\Tuxemon\tuxemon\core\main.py", line 109, in main
    control.main()
  File "C:\Users\Leif\PyCharmProjects\Tuxemon\tuxemon\core\control.py", line 519, in main
    self.main_loop()
  File "C:\Users\Leif\PyCharmProjects\Tuxemon\tuxemon\core\control.py", line 580, in main_loop
    self.update(time_delta)
  File "C:\Users\Leif\PyCharmProjects\Tuxemon\tuxemon\core\control.py", line 203, in update
    state.update(dt)
  File "C:\Users\Leif\PyCharmProjects\Tuxemon\tuxemon\core\states\combat\combat.py", line 155, in update
    self.update_phase()
  File "C:\Users\Leif\PyCharmProjects\Tuxemon\tuxemon\core\states\combat\combat.py", line 332, in update_phase
    self.handle_action_queue()
  File "C:\Users\Leif\PyCharmProjects\Tuxemon\tuxemon\core\states\combat\combat.py", line 344, in handle_action_queue
    self.perform_action(*action)
  File "C:\Users\Leif\PyCharmProjects\Tuxemon\tuxemon\core\states\combat\combat.py", line 563, in perform_action
    if result["name"] == "capture":
KeyError: 'name'

**Comment by presstabstart** [comment]
fixed now.

**Pull request merged**